### ISSUE TITLE
loading data

### ISSUE THREAD

**Issue by sethfrietze** [GuangchuangYu/clusterProfiler#36]
HI Guangchuang,
First, thanks for the great tools. I am determined to use them correctly and frequently!
I'm learning R and I am currently having trouble inputing my own data into clusterProfiler. I can get all functions to work beautifully using geneList example data, however I cannot manage to get my data correctly formatted for this analysis. I have a list of gene symbols and their decreasing fold change expression values, I convert these to entrezid using bitr:
all_gene = scan("all_gene.txt", what="", sep="\n")
all_gene_tx = bitr(all_gene, fromType="SYMBOL", toType="ENTREZID", annoDb="org.Mm.eg.db")
write.csv(all_gene_tx, file="all_gene_tx.csv")
#use vlookup in excel to match up all remapped entrezids to genesymbols and make translated.txt file col1=entrezid, col2=log2 fold-change
tester = read.table("translated.txt", sep="\t", header=T, row.names=1)
head(tester)
                  x
107574    -7.689993
26380     -6.752766
107575    -6.186174
100126433 -5.955670
12323     -5.888340
386750    -5.771180

This is different than the geneList:
head(geneList)
    4312     8318    10874    55143    55388      991 
4.572613 4.514594 4.418218 4.144075 3.876258 3.677857 

so, I try to transpose dataset in excel (renamed to DEG.txt), but the columns get an X added to the names:
dF4_gene <- as.matrix(read.table("DEG.txt", header=TRUE, sep = "\t",
                  row.names = 1,
                  as.is=TRUE))
head (dF4_gene)
                X26380   X107575 X100126433   X12323  X386750    X13390
-7.689993246 -6.752766 -6.186174   -5.95567 -5.88834 -5.77118 -5.749012
                X68070    X67937  X677296   X268663    X12931   X212124
-7.689993246 -5.576549 -5.531749 -5.34585 -5.340453 -5.301863 -5.232462
              X224833    X18585    X18667  X104174  X328795    X12145    X19713
-7.689993246 -5.13658 -5.064886 -5.010068 -4.97236 -4.63558 -4.618544 -4.603771

Basically, how can I go from a 2 column text file with entrezid and expression_change to run enrichment tests, etc in clusterprofiler?

Thanks,
Seth

**Comment by GuangchuangYu** [comment]
see the [wiki](https://github.com/GuangchuangYu/DOSE/wiki/how-to-prepare-your-own-geneList) page.
